File: Models/ClusteringRun.js
import mongoose from 'mongoose';

const clusteringRunSchema = new mongoose.Schema({
  eventId: {
    type: Number,
    default: 1,
  },
  executedAt: {
    type: Date,
    default: Date.now,
  },
  teamsGenerated: {
    type: Number,
    required: true,
  },
  parameters: {
    targetTeamSize: Number,
    maxIterations: Number,
    algorithm: {
      type: String,
      default: 'KMeans_SkillVector',
    },
  },
  teamIds: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Team',
  }],
  status: {
    type: String,
    enum: ['Completed', 'Failed'],
    default: 'Completed',
  },
  durationMs: {
    type: Number,
    default: 0,
  },
});

clusteringRunSchema.index({ eventId: 1, executedAt: -1 });

export const ClusteringRun = mongoose.model('ClusteringRun', clusteringRunSchema);

File: data/Models/ClusteringRun.cs
using System;
using System.Collections.Generic;

namespace TeamWeave.Models
{
    public class ClusteringRun
    {
        public int Id { get; set; }
        public int EventId { get; set; }
        public DateTime ExecutedAt { get; set; }
        public int TeamsGenerated { get; set; }
        public string Parameters { get; set; } // JSON serialized clustering parameters
        public string Status { get; set; } // "Completed", "Failed", etc.

        public virtual ICollection<Team> Teams { get; set; }

        public ClusteringRun()
        {
            ExecutedAt = DateTime.Now;
            Teams = new HashSet<Team>();
            Status = "Completed";
        }
    }
}
